test_pull_requests

```diff
@@ -1189,8 +1388,43 @@ popd
         # comment to reflect the new state
         if submit_test_job
           delete_comment_with_prefix(id, base_repo, ACTION_PREFIX, comments)
-          extended_tests = get_extended_tests(req, comments, branch, settings)
-          submit_tests(repo_to_pull_request, base_repo, branch, comment_id, extended_tests, settings)
+          # Check for pretest_settings_key, so we might skip a round
```

Comment:
I still think the following 40LOC could be broken into a couple of methods to make it read nicer, but that's an opinion